### PR TITLE
Require imageio>=2.33 to fix incompatibility with Pillow>=10.1.0

### LINKED ISSUE #7207: Incompatibility between pillow 10.1.0 and imageio<=2.31.5 breaks roundtrip test
### Description:

See https://github.com/imageio/imageio/issues/1044 for the upstream issue or [this job for our test failure](https://github.com/scikit-image/scikit-image/actions/runs/6534930310/job/17743233190#step:5:625). I'll add a temporary pin to pillow next to fix our CI in the meantime. 

Let's keep this issue open until a patched imageio allows us to revert that pin so we don't forget.

### Way to reproduce:

Ensure pillow 10.1.0 and imageio<=2.31.5 are installed and run
```
$ spin test -- -k imsave_roundtrip
[...]
                # Let pillow know that it is okay to return 16-bit
>               image.mode = desired_mode
E               AttributeError: property 'mode' of 'PngImageFile' object has no setter
```

### PR DESCRIPTION
**Pull request by sitic**
## Description
Pillow release 10.1.0 broke imageio, the new imageio release 2.33.0 fixes the issue ([upstream fix](https://github.com/imageio/imageio/pull/1045)).

Require imageio>=2.33 and revert the hotfix Pillow pin.

Closes #7207 
Reverts #7208 

## Release note

Summarize the introduced changes in the code block below in one or a few sentences. The
summary will be included in the next release notes automatically:

```release-note
Update imageio to fix Pillow incompatibility
```

### CODE REVIEW

**Review by stefanv — APPROVED** [review]

**Comment by sitic** [comment]
The min requirement tests are failing because the new imageio tries to use `PIL.GifImagePlugin.LOADING_STRATEGY`, which is only available for `pillow>=9.1.0`, but the min requirements tests use `Pillow==9.0.1`.

I've filed an imageio bug report
- https://github.com/imageio/imageio/issues/1052

**Comment by mkcor** [comment]
Thanks for looking into this, @sitic!

@stefanv do we want to wait for an [upstream fix](https://github.com/imageio/imageio/issues/1052#issuecomment-1822175295) or pin
```
pillow>=9.1.0,<10.1.0
```
in `requirements/default.txt`?

**Comment by stefanv** [comment]
Do you mean:

```
pillow>=9.1.0
```

? That seems fine.

**Comment by mkcor** [comment]
> Do you mean:
> 
> ```
> pillow>=9.1.0
> ```
> 
> ? That seems fine.

Oops, yes, sorry! @sitic do you mind updating the PR accordingly?

File: pyproject.toml [hunk 1]
```diff
@@ -28,8 +28,8 @@ dependencies = [
     'numpy>=1.22',
     'scipy>=1.8',
     'networkx>=2.8',
-    'pillow>=9.0.1,<10.1.0',
-    'imageio>=2.27',
+    'pillow>=9.0.1',
```
Comment (jarrodmillman) [inline pyproject.toml:31]:
```suggestion
    'pillow>=9.1',
```

File: requirements/default.txt [hunk 2]
```diff
@@ -3,8 +3,8 @@
 numpy>=1.22
 scipy>=1.8
 networkx>=2.8
-pillow>=9.0.1,<10.1.0
-imageio>=2.27
+pillow>=9.0.1
```
Comment (jarrodmillman) [inline requirements/default.txt:6]:
```suggestion
pillow>=9.1
```

**Review by jarrodmillman — CHANGES_REQUESTED** [review]

**Review by mkcor — APPROVED** [review]
:shipit:

**Review by jarrodmillman — APPROVED** [review]

**Pull request merged**